include/aws/s3/private/s3_request.h

```diff
@@ -122,8 +122,19 @@ struct aws_s3_request_metrics {
         int error_code;
         /* Retry attempt. */
         uint32_t retry_attempt;
+        /* Is the memory for the request allocated from the buffer pool or not. */
+        bool memory_allocated_from_pool;
     } crt_info_metrics;
 
+    struct {
```

Comment:
thinking out loud:
maybe we should unify this with https://github.com/awslabs/aws-c-s3/blob/332dd22c47a7ed139eee71e7f219b764ef8cdf4c/include/aws/s3/private/s3_meta_request_impl.h#L297. 
we can probably do some of the same sanity checks in upload path as well